File: Javascript/SymmetricTree.js
// Leetcode Problem - 101

function TreeNode(val,left,right){
    this.val = (val === undefined ? 0 : val)
    this.left = (left === undefined ? null : left)
    this.right = (right === undefined ? null : right)
}

var isSymmetric = function(root){

    const queue = []

    queue.push(root.left)
    queue.push(root.right)

    while (queue.length>0){
        lChild = queue.shift()
        rChild = queue.shift()

        if (!lChild && !rChild){
            continue
        }else if ((!lChild || !rChild) || (lChild.val !== rChild.val)){
            return false
        }

        queue.push(lChild.left)
        queue.push(rChild.right)
        queue.push(lChild.right)
        queue.push(rChild.left)
    }

    return true
}

File: Python/SymmetricTree.py
## Leetcode Problem - 101

class TreeNode:
    def __init__(self, val=0, left=None, right=None):
        self.val = val
        self.left = left
        self.right = right

def isSymmetric(root):
    
    def dfs(lChild,rChild):
        if not lChild and not rChild:
            return True
        if not lChild or not rChild:
            return False
        
        return (lChild.val == rChild.val \
            and dfs(lChild.left,rChild.right) \
            and dfs(lChild.right,rChild.left))

    return dfs(root.left,root.right)

root = [1,2,2,3,4,4,3]
print(isSymmetric(root))